hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/discovery/DiscoveryAddressTranslator.java

```diff
@@ -61,7 +61,7 @@ public Address translate(Address address) {
 
         privateToPublic = this.privateToPublic;
         Address publicAddress = privateToPublic.get(address);
-        if (!alreadyRefreshed) {
+        if (publicAddress == null && !alreadyRefreshed) {
```

Comment:
OK, I thought that we wanted force the second refresh to force finding the public ip, but it probably will not end up with a different result from the second refresh.